File: src/MinimalWebHooks.Core/Models/DbSets/WebhookClient.cs
using System.ComponentModel.DataAnnotations;
using MinimalWebHooks.Core.Enum;

namespace MinimalWebHooks.Core.Models.DbSets;

public class WebhookClient
{
    public int Id { get; set; }
    [Required]
    public string Name { get; set; }
    [Required]
    public string WebhookUrl { get; set; }
    [Required]
    public WebhookActionType ActionType { get; set; }
    [Required]
    public string EntityTypeName { get; set; }
    public bool Disabled { get; set; }
    public List<WebhookClientHeader>? ClientHeaders { get; set; }

    public bool HasHeaders() => ClientHeaders != null && ClientHeaders.Any();
}

File: example/Admin/src/types/webhooks/webhook-client.ts
import {
  WebhookClientActivityLog,
} from '@/types/webhooks/webhook-client-activity-log';
import { WebhookClientHeader } from '@/types/webhooks/webhook-client-header';

export type WebhookClient = {
    id: number,
    name: string,
    webhookUrl: string,
    actionType: number,
    entityTypeName: string,
    disabled: boolean,
    clientHeaders?: WebhookClientHeader[],
    activityLogs?: WebhookClientActivityLog[]
}
